kwalitee/repo.py

```diff
@@ -35,63 +42,60 @@ def clone(self):
+    def get_latest_version(self):
         with pushd_popd(self.path):
-            ret['number_of_commits'] = self._run('git rev-list HEAD --count').stdout
-            ret['last_commit'] = self._run(r'git log -1 --format=%cd ').stdout
-            ret['url'] = self._run('git config --get remote.origin.url').stdout
-            ret['latest_tag'] = self._run('git describe --abbrev=0 --tags').stdout
-        return ret
+            return self._run('git describe --abbrev=0 --tags').stdout
 
-    def get_file_contents(self):
+    def make_additional_info(self):
+        result = {}
+        result['links'] = self.get_file_urls()
+        #result['description'] = self.get_description()
+
+        return result
+
+
+    def get_file_urls(self):
         ret = {}
-        self.log.info("%s  Getting file contents" % self.url)
+        self.log.info("%s  Getting file URLs" % self.url)
         with pushd_popd(self.path):
             for path in [Path(x) for x in ['ocrd-tool.json', 'Dockerfile', 'README.md', 'setup.py']]:
                 if path.is_file():
-                    with path.open() as f:
-                        ret[path.name] = f.read()
+                    ret[path.name] = self.url + '/blob/master/' + str(path)
                 else:
                     ret[path.name] = None
         return ret
 
-    def get_python_info(self):
-        ret = {}
+    def get_description(self):
+        api_url = "https://api.github.com/repos/" + self.org_plus_name
+        header = {"Accept": "application/vnd.github.v3+json"}
+        response = requests.get(api_url, headers=header)
+        response_json = json.loads(response.text)
+
+        return response_json['description']
+
+    def validate_ocrd_tool_json(self):
+        valid = False
         with pushd_popd(self.path):
-            ret['url'] = self._run('python3 setup.py --url').stdout
-            ret['name'] = self._run('python3 setup.py --name').stdout
-            ret['author'] = self._run('python3 setup.py --author').stdout
-            ret['author-email'] = self._run('python3 setup.py --author-email').stdout
-        self.log.info("  Fetching pypi info")
-        response = R.get('https://pypi.python.org/pypi/%s/json' % ret['name'])
-        ret['pypi'] = json.loads(response.text) if response.status_code == 200 else None
-        return ret
+            if Path('ocrd-tool.json').is_file():
+                result = self._run('ocrd ocrd-tool ocrd-tool.json validate').stdout
+                if 'valid="true"' in result:
+                    valid = True
+        return valid
 
-    def get_ocrd_tools(self):
-        ot = json.loads(self.get_file_contents()['ocrd-tool.json'])
-        return ot['tools']
+    def get_project_type(self):
+        type = 'bashlib'
+        with pushd_popd(self.path):
+            for path in [Path(x) for x in ['setup.py', 'requirements.txt', 'requirements_test.txt']]:
+                if path.is_file():
+                    type = 'python'
```

Comment:
Mmh, I'm not sure if this is sufficient. Your suggestion would result in `core` being identified as `bashlib`, because it is built with a makefile instead of a `setup.py`. That's also why I have this `requirements_test.txt` in there–it's only for `core`.